pyrelational/datasets/uci_datasets.py

```diff
@@ -0,0 +1,194 @@
+            "energy": "http://archive.ics.uci.edu/ml/machine-learning-databases/00242/ENB2012_data.xlsx",
+            "power": "https://archive.ics.uci.edu/ml/machine-learning-databases/00294/CCPP.zip",
+            "wine": "https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/winequality-red.csv",
+            "yacht": "http://archive.ics.uci.edu/ml/machine-learning-databases/00243/yacht_hydrodynamics.data",
+            "glass": "https://archive.ics.uci.edu/ml/machine-learning-databases/glass/glass.data",
+            "parkinsons": "https://archive.ics.uci.edu/ml/machine-learning-databases/parkinsons/parkinsons.data",
+            "seeds": "https://archive.ics.uci.edu/ml/machine-learning-databases/00236/seeds_dataset.txt",
+            "airfoil": "https://archive.ics.uci.edu/ml/machine-learning-databases/00291/airfoil_self_noise.dat",
+        }
+        self.data_dir = data_dir
+        self.name = name
+        self.n_splits = n_splits
+
+        # flag for classification/regression dataset
+        if name in ["glass", "parkinsons", "seeds"]:
+            self.classification = True
+        else:
+            self.classification = False
+
+        self._load_dataset()
+
+    def _load_dataset(self):
+        if self.name not in self.datasets:
+            raise Exception("Not part of datasets supported in PyRelationAL at the moment")
+        if not path.exists(self.data_dir):
+            os.mkdir(self.data_dir)
+        if not path.exists(self.data_dir + "UCI"):
+            os.mkdir(self.data_dir + "UCI")
+
+        url = self.datasets[self.name]
+        file_name = url.split("/")[-1]
+        if not path.exists(self.data_dir + "UCI/" + file_name):
+            urllib.request.urlretrieve(self.datasets[self.name], self.data_dir + "UCI/" + file_name)
+        data = None
+
+        if self.name == "housing":
+            data = pd.read_csv(self.data_dir + "UCI/housing.data", header=0, delimiter=r"\s+").values
+            self.data = data[np.random.permutation(np.arange(len(data)))]
+
+        elif self.name == "airfoil":
+            data = pd.read_csv(self.data_dir + "UCI/airfoil_self_noise.dat", header=0, delimiter=r"\s+").values
+            self.data = data[np.random.permutation(np.arange(len(data)))]
+
+        elif self.name == "concrete":
+            data = pd.read_excel(self.data_dir + "UCI/Concrete_Data.xls", header=0).values
+            self.data = data[np.random.permutation(np.arange(len(data)))]
+
+        elif self.name == "energy":
+            data = pd.read_excel(self.data_dir + "UCI/ENB2012_data.xlsx", header=0).values
+            self.data = data[np.random.permutation(np.arange(len(data)))]
+
+        elif self.name == "power":
+            zipfile.ZipFile(self.data_dir + "UCI/CCPP.zip").extractall(self.data_dir + "UCI/")
+            data = pd.read_excel(self.data_dir + "UCI/CCPP/Folds5x2_pp.xlsx", header=0).values
+            np.random.shuffle(data)
```

Comment:
Why the shuffle here?

Reply:
You are right this is redundant. Fixed in latest commit.